### PR TITLE
ATLAS-196: Distinguish between web and REST endpoints using an api folder

### PR DESCRIPTION
**Pull request by HelioStrike**
JIRA issue: https://issues.openmrs.org/browse/ATLAS-196

### CODE REVIEW

**Comment by bmamlin** [comment]
Were you planning on doing this is stages? I was expecting not only to move the files, but also to move the endpoints (routes), so all REST API calls would be under `/api/*`. For example, fetching a marker would use `GET /api/marker/:id`, which would mean changing all URLs using the REST API and updating the routes mapped for API routes.

**Comment by HelioStrike** [comment]
@bmamlin Ohh, I thought that the URLs would remain the same. Its getting late now, so I'll update the PR tomorrow.

**Comment by HelioStrike** [comment]
@bmamlin Updated the PR :)

**Comment by HelioStrike** [comment]
@bmamlin Updated the PR. I think everything's working but there were 2-3 iterations where I thought everything was done and then I find loose ends. :p

File: public/js/user.js [hunk 1]
```diff
@@ -383,15 +383,15 @@ function saveMarker(e) {
...
+      url: "/api/marker/"+id,
       type: "PATCH",
       data: json,
       dataType: "json",
       processData: false,
       contentType: "application/json",
     })
         .done(function(response) {
-          sites[id].siteData = response;
+          sites[id].siteData.date_changed = response.date_changed;
```
Comment (HelioStrike) [inline public/js/user.js]: This isn't related to the issue, but I found that updating a site and then editing it throws an error 'site.auth' not found. Here's a quick fix. :)

**Comment by HelioStrike** [comment]
@bmamlin I've been testing the PR and I haven't found any bugs in a while.

**Comment by bmamlin** [comment]
Looks good. Behaved well when I tested, so merged it. Great work, @HelioStrike!

**Pull request merged**